Public/Src/Tools/SymbolDaemon/Tool.SymbolDaemonRunner.dsc

```diff
@@ -165,15 +168,55 @@ function addFiles(createResult: SymbolCreateResult, args: OperationArguments, fi
         uberArgs,
         ipcArgs => ipcArgs.merge<Transformer.IpcSendArguments>({
             messageBody: [
-                ...fileMessageBody,                    
+                ...fileMessageBody,
             ],
             lazilyMaterializedDependencies: [
-                ...files,                    
+                // note: symbolMetadataFile is not included in this list
+                ...files,
             ],
         })
     );
 }
 
+function indexSymbolFiles(files: File[]) : DerivedFile {
+    const symbolDataFileName = "symbol_data.txt";
+
+    Contract.requires(
+        files.length !== 0,
+        "The list of files cannot be empty"
+    );   
+
+    // get a directory where we are going to store the result of indexing
+    const outDir = Context.getNewOutputDirectory("symbol_indexing");

```

Comment:
>getNewOutputDirectory [](start = 27, length = 21)

Cool. I guess this already does the right thing since it takes the stack frame of the requesting evaluated value into account for uniquness

Reply:
Yeah, I had sorting/hashing of inputs, and used that hash as an input into getNewOutputDirectory. However, Lance pointed out that this is an unnecessary step that can be avoided, and bxl will handle any collisions

---
In reply to: [327694554](https://github.com/microsoft/BuildXL/pull/921#discussion_r327694554) [](ancestors = 327694554)